client/blocks/upload-image/index.jsx

```diff
@@ -250,7 +255,12 @@ class UploadImage extends Component {
 	}
 
 	removeUploadedImage = () => {
+		const { onUploadedImageRemove } = this.props;
+		const { uploadedImage } = this.state;
+
 		this.setState( { uploadedImage: null } );
```

Comment:
All of these gets either re-written upon new image selection/uploading or cleared upon unmounting the component. 

However, we are not revoking the object URLs which could cause memory leaks. Great catch!